core/arch/arm/kernel/generic_timer.c

```diff
@@ -42,13 +42,15 @@ __relrodata_unpaged("timer_desc") = {
 	.get_now = timer_now,
 	.is_per_cpu = true,
 };
+DECLARE_KEEP_PAGER(timer_desc);
```

Comment:
Isn't `timer_desc` already unpaged since it has `__relrodata_unpaged("timer_desc")`? In which case it could be the `is_unpaged()` function that needs fixing?

Reply:
`__relrodata_unpaged("timer_desc")` shouldn't be needed any longer with `DECLARE_KEEP_PAGER(timer_desc)`. The latter makes all dependencies (including the callback functions) unpaged. `__relrodata_unpaged("timer_desc")` only affects "timer_desc" itself.

Reply:
It turns out I was wrong, `__relrodata_unpaged("timer_desc")` is indeed needed.

Reply:
Why is it? My understanding is aligned with your statement that `DECLARE_KEEP_PAGER()` should do the job.

Reply:
It's not enough in the virtualization case.

Reply:
Ok, i'll keep that in mind. Thanks.